joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.h

```diff
@@ -246,8 +246,41 @@ class JointTrajectoryController : public controller_interface::Controller<Hardwa
    */
   void setHoldPosition(const ros::Time& time, RealtimeGoalHandlePtr gh=RealtimeGoalHandlePtr());
 
+  /**
+   * \brief Internal update function.
+   *
+   * Update time data, current states and errors and check tolerances.
+   */
+  void updateStateAndTimeData(const ros::Time& time, const ros::Duration& period);
```

Comment:
Overriding in derived classes will need some `virtual`s here

Reply:
Thanks! Done.